bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java

```diff
@@ -0,0 +1,836 @@
+
+            final ThingBuilder thingBuilder = editThing();
+            thingBuilder.withProperties(newProperties);
+            updateThing(thingBuilder.build());
+        }
+        return updatedBinding;
+    }
+
+    private synchronized boolean wasThingUpdatedExternally(BondDevice devInfo) {
+        // Check if the Bond hash tree has changed
+        config = getConfigAs(BondDeviceConfiguration.class);
+        final String lastDeviceConfigurationHash = config.lastDeviceConfigurationHash;
+        boolean updatedHashTree = !devInfo.hash.equals(lastDeviceConfigurationHash);
+        if (updatedHashTree) {
+            logger.info("Hash tree of device has been updated by Bond.");
+            logger.info("Current state is {}, prior state was {}.", devInfo.hash, lastDeviceConfigurationHash);
+        }
+        return updatedHashTree;
+    }
+
+    private boolean getBridgeAndAPI() {
+        Bridge myBridge = this.getBridge();
+        if (myBridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "No Bond bridge is associated with this Bond device");
+
+            return false;
+        } else {
+            BondBridgeHandler myBridgeHandler = (BondBridgeHandler) myBridge.getHandler();
+            if (myBridgeHandler != null) {
+                this.api = myBridgeHandler.getBridgeAPI();
+                return true;
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Cannot access API for Bridge associated with this Bond device");
+                return false;
+            }
+        }
+    }
+
+    private void setBridgeOffline(ThingStatusDetail detail, String description) {
+        Bridge myBridge = this.getBridge();
+        if (myBridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "No Bond bridge is associated with this Bond device");
+            logger.error("No Bond bridge is associated with this Bond device - cannot create device!");
+            return;
+        } else {
+            BondBridgeHandler myBridgeHandler = (BondBridgeHandler) myBridge.getHandler();
+            if (myBridgeHandler != null) {
+                myBridgeHandler.setBridgeOffline(detail, description);
+                return;
+            }
+        }
+    }
+
+    // Start polling for state
+    private synchronized void startPollingJob() {
+        final ScheduledFuture<?> pollingJob = this.pollingJob;
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            Runnable pollingCommand = () -> {
+                BondHttpApi api = this.api;
+                if (api == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Bridge API not available");
+                    initialize();
```

Comment:
It should not be needed to call `initialize` here. If some code in `initialize` needs to be called this should be put into a separate method. `initialize` is generally called when the thing itself is initialized by openHAB. So the binding should not do that itself.